src/services/UserService.ts

```diff
@@ -96,31 +95,61 @@ class UserService {
   /**
    * Registers a user in Firebase and in the BE
    *
-   * @param firebase - The firebase object
+   * @param firebaseService - The firebaseService object
    * @param data - Data to register the user with
+   * todo: pass in AuthUserContext
+   * todo update AuthUserContext with the user id
    */
-  public async registerUser(firebase: Firebase, data: UserInputData): Promise<void> {
-    return firebase
-      .doCreateUserWithEmailAndPassword(data.email, data.password)
-      .then(() => {
-        // TODO: Register user in backend
-        // https://promopal.atlassian.net/browse/PP-36
-        // https://promopal.atlassian.net/browse/PP-38
-      })
-      .catch((error: Error) => {
-        return Promise.reject(error);
-      });
+  public async registerUser(firebaseService: FirebaseService, data: UserInputData): Promise<void> {
+    try {
+      const userCredential: firebase.auth.UserCredential = await firebaseService.doCreateUserWithEmailAndPassword(
+        data.email,
+        data.password
+      );
```

Comment:
If I uncomment the whole function except this chunk, it will direct me to the user details account page. My best guess is this is because firebaseService is used in a provider, I think once we remove that provider it should go away